arangod/MMFiles/MMFilesCollection.cpp

```diff
@@ -2821,14 +2811,15 @@ Result MMFilesCollection::truncate(transaction::Methods& trx, OperationOptions&
   }
 
   READ_LOCKER(guard, _indexesLock);
-  auto indexes = _indexes;
-  size_t const n = indexes.size();
-
   TRI_voc_tick_t tick = TRI_NewTickServer();
-  for (size_t i = 1; i < n; ++i) {
-    auto idx = indexes[i];
-    TRI_ASSERT(idx->type() != Index::IndexType::TRI_IDX_TYPE_PRIMARY_INDEX);
-    idx->afterTruncate(tick);
+  if (_indexes.size() > 1) {
+    auto indexes = _indexes;
```

Comment:
why are we taking a copy of _indexes here? I think we can work directly on `_indexes`.